packages/shylu/shylu_dd/frosch/src/CoarseSpaces/FROSch_CoarseSpace_def.hpp

```diff
@@ -138,16 +139,21 @@ namespace FROSch {
                             FROSCH_ASSERT(NumVectors_i+itmp <= AssembledBasis_->getNumVectors(),"FROSch::CoarseSpace: NumVectors_i+itmp <= AssembledBasis_->getNumVectors()");
                             FROSCH_ASSERT(LocalLength_i+Offsets_[i] <= AssembledBasis_->getLocalLength(),"FROSch::CoarseSpace: LocalLength_i+Offsets_[i] <= AssembledBasis_");
 
+                            Kokkos::RangePolicy<execution_space> policy (0, LocalLength_i);
+                            // Xpetra wrapper for Tpetra MV
+                            auto unassembledXTpetraMVector = rcp_dynamic_cast<const TpetraMultiVector<SC,LO,GO,NO>>(UnassembledSubspaceBases_[i]);
```

Comment:
It might make sense to use `rcp_dynamic_cast<..>(..,True)` since you don't want to accept a `null` return.